corehq/apps/registration/utils.py

```diff
@@ -138,19 +134,20 @@ def request_new_domain(request, form, is_new_user=True):
     if is_new_user:
         dom_req.save()
         if settings.IS_SAAS_ENVIRONMENT:
+            #  Load template apps to the user's new domain in parallel
             from corehq.apps.app_manager.tasks import load_appcues_template_app
-            chain(
-                load_appcues_template_app.si(new_domain.name, current_user.username, HEALTH_APP),
-                load_appcues_template_app.si(new_domain.name, current_user.username, AGG_APP),
-                load_appcues_template_app.si(new_domain.name, current_user.username, WASH_APP),
-                send_domain_registration_email.si(
-                    request.user.email,
-                    dom_req.domain,
-                    dom_req.activation_guid,
-                    request.user.get_full_name(),
-                    request.user.first_name
-                )
-            ).apply_async()
+            header = [
+                load_appcues_template_app.si(new_domain.name, current_user.username, slug)
+                for slug in APPCUES_APP_SLUGS
+            ]
+            callback = send_domain_registration_email.si(
+                request.user.email,
+                dom_req.domain,
+                dom_req.activation_guid,
+                request.user.get_full_name(),
+                request.user.first_name
+            )
+            chord(header)(callback)
```

Comment:
Such funny syntax :) Does this work okay? In some of the examples it looked like the list of results of the header tasks was passed in as an argument to the callback. Just to check that this works without having to change the signature of the `send_domain_registration_email` function.

Reply:
I do see what you mean about the callback taking the results of the functions in the header, but the apps still load correctly and before the email is sent. What does your gut tell you could be going wrong?

Reply:
Yeah if you locally make sure `CELERY_ALWAYS_EAGER` is `False`, stop celery, start celery, and then test this through the UI then it's probably fine.

The thing I was wondering about was whether the callback task (`send_domain_registration_email`) needed to take an extra argument (the list of results of the headers) for this to work. But if it works then I guess not.